pilot/docker/Dockerfile.proxytproxy

```diff
@@ -0,0 +1,43 @@
+FROM istionightly/base_debug
+ARG proxy_version
+ARG istio_version
+
+# Please keep in sync with Dockerfile.v2
+# This proxy will use TPROXY interception. Core dump won't work - see issue 5745
+# User will need to use the proxy annotation to use this image, as well as the
+# original InterceptionMode annotation to run as root and add NET_ADMIN
+
+# TODO: evaluate if we should add NET_ADMIN in all cases, to support future firewall
+# and dynamic iptable config
+
+ADD envoy /usr/local/bin/envoy
+
+ADD pilot-agent /usr/local/bin/pilot-agent
+
+# pilot-agent and envoy may run with effective uid 0 in order to run envoy with
+# CAP_NET_ADMIN, so any iptables rule matching on "-m owner --uid-owner
+# istio-proxy" will not match connections from those processes anymore.
+# Instead, rely on the process's effective gid being istio-proxy and create a
+# "-m owner --gid-owner istio-proxy" iptables rule in istio-iptables.sh.
+RUN \
+  chgrp 1337 /usr/local/bin/envoy /usr/local/bin/pilot-agent && \
+  chmod 2755 /usr/local/bin/envoy /usr/local/bin/pilot-agent
+
+ADD envoy_pilot.yaml.tmpl /etc/istio/proxy/envoy_pilot.yaml.tmpl
+ADD envoy_policy.yaml.tmpl /etc/istio/proxy/envoy_policy.yaml.tmpl
+ADD envoy_telemetry.yaml.tmpl /etc/istio/proxy/envoy_telemetry.yaml.tmpl
+ADD istio-iptables.sh /usr/local/bin/istio-iptables.sh
+
+COPY envoy_bootstrap_v2.json /var/lib/istio/envoy/envoy_bootstrap_tmpl.json
+
+ENV ISTIO_META_ISTIO_PROXY_SHA $proxy_version
+ENV ISTIO_META_ISTIO_VERSION $istio_version
+ENV ISTIO_META_INTERCEPTION_MODE TPROXY
```

Comment:
That’s not necessary here, since that will be set by the injector.

Reply:
Actually, this should NOT be set to TPROXY if the other conditions are not met (run the sidecar with CAP_NET_ADMIN, in particular), otherwise Envoy will just keep crashing trying to set the IP_TRANSPARENT socket option on the listener.
So you really should remove this ENV from the Dockerfile.

Reply:
My plan is to make the init container customizable and identical with the sidecar container, and pass the same ENV on both. For 1.1 we may provide dynamic changes to the iptable capture. 

The other 'conditions' should go away - if you specify tproxy sidecar (and init) image, it will take care of both and no extra customizations needed. This will also allow to clean up the iptable initializer, which is a mess right now - we'll have a script for tproxy and one for redirect. 

Short term (1.0) - users will need to specify both image and the old annotation. Better then not working at all..

Reply:
> if you specify tproxy sidecar (and init) image, it will take care of both and no extra customizations needed

That's not enough. You need to run istio-proxy with CAP_NET_ADMIN for TPROXY to work. And that can only be done by modifying the pod spec. If you don't want that, you'll have to always run istio-proxy with CAP_NET_ADMIN. I'm fine with that, but I got push back when I had proposed that.

Reply:
Yes, I know - but requirements change. There are plans to have more flexibility in how we capture, which will need NET_ADMIN anyways ( proxy-agent may change iptables at runtime ). We still want to start proxy-agent as root 
and exec envoy as 1337 - I still don't think it's ok to run envoy as root (as it happens in tproxy mode).